src/interactions/panZoomInteraction.ts

```diff
@@ -439,6 +577,112 @@ namespace Plottable.Interactions {
+      }
+      this._minDomainValues.set(quantitativeScale, minDomainValue);
+      return this;
+    }
+
+    /**
+     * Gets the maximum domain value for the scale, constraining the pan/zoom
+     * interaction to a maximum value in the domain.
+     *
+     * Note that this differs from minDomainExtent/maxDomainExtent, in that
+     * those methods provide constraints such as showing at least 2 but no more
+     * than 5 values at a time.
+     *
+     * By contrast, minDomainValue/maxDomainValue set a boundary beyond which
+     * the user cannot pan/zoom.
+     *
+     * @param {QuantitativeScale<any>} quantitativeScale The scale to query
+     * @returns {D} The maximum domain value for the scale.
+     */
+    public maxDomainValue<D>(quantitativeScale: QuantitativeScale<D>): D;
+    /**
+     * Sets the maximum domain value for the scale, constraining the pan/zoom
+     * interaction to a maximum value in the domain.
+     *
+     * Note that this differs from minDomainExtent/maxDomainExtent, in that
+     * those methods provide constraints such as showing at least 2 but no more
+     * than 5 values at a time.
+     *
+     * By contrast, minDomainValue/maxDomainValue set a boundary beyond which
+     * the user cannot pan/zoom.
+     *
+     * @param {QuantitativeScale<any>} quantitativeScale The scale to query
+     * @param {D} maxDomainExtent The maximum domain value for the scale.
+     * @returns {Interactions.PanZoom} The calling PanZoom Interaction.
+     */
+    public maxDomainValue<D>(quantitativeScale: QuantitativeScale<D>, maxDomainValue: D): this;
+    public maxDomainValue<D>(quantitativeScale: QuantitativeScale<D>, maxDomainValue?: D): any {
+      if (maxDomainValue == null) {
+        return this._maxDomainValues.get(quantitativeScale);
+      }
+      let minValueForScale = this.minDomainValue(quantitativeScale);
+      if (minValueForScale != null && maxDomainValue.valueOf() <= minValueForScale.valueOf()) {
+        throw new Error("maxDomainValue must be larger than minDomainValue for the same Scale");
+      }
+      this._maxDomainValues.set(quantitativeScale, maxDomainValue);
+      return this;
+    }
+
+    /**
+     * Uses the current domain of each scale to apply a minimum and maximum
+     * domain value for that scale.
+     *
+     * This constrains the pan/zoom interaction to show no more than the domain
+     * of the scale.
+     */
+    public constrainToScaleDomainValues() {
+      this.xScales().forEach((scale) => this._constrainToScaleDomainValues(scale));
+      this.yScales().forEach((scale) => this._constrainToScaleDomainValues(scale));
+    }
+
+    private _constrainToScaleDomainValues<D>(scale: QuantitativeScale<D>) {
+      this._minDomainValues.delete(scale);
+      this._maxDomainValues.delete(scale);
+      const [ domainMin, domainMax ] = scale.domain();
+      this.minDomainValue(scale, domainMin);
+      this.maxDomainValue(scale, domainMax);
+    }
```

Comment:
this file is getting really large - over 700 lines. We should refactor this soon (would be good to revisit the codebase as a whole here)